pkg/reconciler/pipelinerun/resources/validate_params.go

```diff
@@ -46,3 +47,26 @@ func ValidateParamTypesMatching(p *v1beta1.PipelineSpec, pr *v1beta1.PipelineRun
 	}
 	return nil
 }
+
+// ValidateExpectedParametersProvided validate that all the parameters expected by the Pipeline are provided by the PipelineRun. 
+func ValidateExpectedParametersProvided(p *v1beta1.PipelineSpec, pr *v1beta1.PipelineRun) error {
+	// Build a map of parameter names/types declared in pr.
+	providedParams := make(map[string]v1beta1.ParamType)
+	for _, param := range pr.Spec.Params {
+		providedParams[param.Name] = param.Value.Type
+	}
+
+	// Build a list of parameter names in p that are missing from pr.
+	var missingParams []string
+	for _, param := range p.Params {
+		if _, ok := providedParams[param.Name]; !ok {
+			missingParams = append(missingParams, param.Name)
+		}
+	}
```

Comment:
you might be able to use DiffLeft from https://github.com/tektoncd/pipeline/blob/master/pkg/list/diff.go

Reply:
So useful, thanks!